### PR TITLE
Add test for updated CA bundle

### PR DESCRIPTION
**Pull request by MorrisJobke**
For #14369

### CODE REVIEW

**Comment by kesselb** [comment]
https://curl.haxx.se/docs/caextract.html

I guess they are not to happy if you fetch the file multiple times a day. There are some recommendations at "Automated downloads from here".

**Comment by MorrisJobke** [comment]
> I guess they are not to happy if you fetch the file multiple times a day. There are some recommendations at "Automated downloads from here".

Problem is that the mtime is "now" when doing a fresh git clone. Let me have a look into this.

**Review by rullzer — APPROVED** [review]

**Comment by MorrisJobke** [comment]
> > I guess they are not to happy if you fetch the file multiple times a day. There are some recommendations at "Automated downloads from here".
> 
> Problem is that the mtime is "now" when doing a fresh git clone. Let me have a look into this.

Any other ideas? :/

**Comment by kesselb** [comment]
- Mirror cacert.pem and cacert.pem.sha256 to *.nextcloud.com and fetch it from there?
- Run only if branch name is `version/noid/*`? (not sure if possible but should be available https://0-8-0.docs.drone.io/environment-reference/)
- Write a cacertbot (could be useful for other projects)

**Review by MorrisJobke — COMMENTED** [review]
Makes sense 👍

File: autotest-checkers.sh [hunk 1]
```diff
@@ -8,7 +8,7 @@ php ./build/translation-checker.php
 RESULT=$(($RESULT+$?))
 php ./build/htaccess-checker.php
 RESULT=$(($RESULT+$?))
-php ./build/ca-bundle-checker.sh
+bash ./build/ca-bundle-checker.sh
```
Comment (kesselb) [inline autotest-checkers.sh]: I did not understand why the script was not executed :rofl: :see_no_evil:

**Comment by kesselb** [comment]
image

https://drone.nextcloud.com/nextcloud/server/16761/86 (branch name: ci/noid/add-bundle-test) :heavy_check_mark:

File: build/ca-bundle-checker.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+echo
+if [[ -n ${DRONE_BRANCH} &&  ! ${DRONE_BRANCH} =~ version\/noid\/.+ ]]; then
+    echo "Skip CA bundle check"
+    exit 1
```
Comment (MorrisJobke) [inline build/ca-bundle-checker.sh]: Should be `0` 🙈

Comment (kesselb) [reply build/ca-bundle-checker.sh]: Of course.

**Comment by MorrisJobke** [comment]
Bundle properly skipped - let me rebase for squash all the commits in here.

**Comment by MorrisJobke** [comment]
@kesselb Ready for final review 👍

**Comment by kesselb** [comment]
https://drone.nextcloud.com/nextcloud/server/16769/86

Looks like $DRONE_BRANCH is the target branch. We can either check against DRONE_REFSPEC for version/noid/ or check if DRONE_BRANCH is stable(\d){2}.

**Comment by kesselb** [comment]
> DRONE_BRANCH is stable(\d){2}.

Not an option because this is true for every backport ... :imp:

**Comment by MorrisJobke** [comment]
Then let's merge it.


👍 From my side for your code.

**Pull request merged**